malboxes/malboxes.py

```diff
@@ -212,44 +274,60 @@ def run_foreground(command):
         return p.returncode
 
 
-def run_packer(packer_config):
+def run_packer(packer_tmpl, args):
     print("Starting packer to generate the VM")
     print("----------------------------------")
 
-    # packer or packer-io?
-    binary = 'packer'
-    if shutil.which(binary) == None:
-        binary = 'packer-io'
+    prev_cwd = os.getcwd()
+    os.chdir(DIRS.user_cache_dir)
+
+    try:
+        # packer or packer-io?
+        binary = 'packer'
         if shutil.which(binary) == None:
-            print("packer not found. Install it: "
-                  "https://www.packer.io/intro/getting-started/setup.html")
-            return 254
-
-    # run packer with relevant config
-    configfile = os.path.join(DIRS.user_config_dir, 'config.json')
-    cmd = [binary, 'build',
-           '-var-file={}'.format(configfile),
-           "-var", "malboxes_cache_dir={}".format(DIRS.user_cache_dir),
-           packer_config]
-    ret = run_foreground(cmd)
+            binary = 'packer-io'
+            if shutil.which(binary) == None:
+                print("packer not found. Install it: "
+                      "https://www.packer.io/intro/getting-started/setup.html")
+                return 254
```

Comment:
why not ;)

the idea is exit code: 2, 3, 4 is stuff we control. For stuff we don't control (like binary isn't there) I decided to start with the largest number.

I'll extract those to constants which should document the error number better but not in that branch.